### PR TITLE
Peer banning reason

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

PR adds in a way for keep track in the peer database why a peer was banned. 

## Motivation and Context

Currently the peer db only keeps track of if they are banned and until when. This adds in a a reason why the peer was banned. This will allow retrospective inspection of why peers are getting banned

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: comms/src/peer_manager/peer.rs [hunk 1]
```diff
@@ -71,6 +71,7 @@ pub struct Peer {
     /// Flags for the peer.
     pub flags: PeerFlags,
     pub banned_until: Option<NaiveDateTime>,
+    pub banned_reason: String,
```
Comment (sdbondi) [inline comms/src/peer_manager/peer.rs:74]: This is a breaking change - we can write migrations for this (the infrastructure is there - see [here](https://github.com/tari-project/tari/blob/690b9ee09750c8e8f4d44c1a916a476c082a727f/comms/src/peer_manager/migrations/v1.rs#L68)) so users (esp mobile) are not disrupted, or we need to require that peer DBs are deleted.

**Review by sdbondi — COMMENTED** [review]
Happy with this change - it is a breaking change for the peer db so I think writing a migration is the path of least resistance (esp for mobile).

**Comment by delta1** [comment]
I'm getting some compile errors in comms when running `cargo test` eg: https://github.com/SWvheerden/tari/blob/sw_add_ban_reason/comms/src/peer_manager/manager.rs#L329

**Comment by SWvheerden** [comment]
> I'm getting some compile errors in comms when running `cargo test` eg: https://github.com/SWvheerden/tari/blob/sw_add_ban_reason/comms/src/peer_manager/manager.rs#L329

haha I know what happend...
I never compiled the test modules

**Review by stringhandler — COMMENTED** [review]
Happy with the code. Is it outputted anywhere? If you can add it to the `list-peers` command I'm happy to merge it

**Comment by SWvheerden** [comment]
> outputted anywhere? If you can add it to the `list-peers` command I'm happy to merge it

Its added to the peer display

**Review by stringhandler — APPROVED** [review]
Sorry, must have missed that

**Pull request merged**